balena/models/device.py

```diff
@@ -809,51 +860,67 @@ def __restart_application():
 
         with_supervisor_locked_error(__restart_application)
 
-    def reboot(self, uuid_or_id: Union[str, int], force: bool = False) -> None:
+    def reboot(self, uuid_or_id: Optional[Union[str, int]] = None, force: bool = False) -> None:
         """
         Reboot the device.
 
         Args:
-            uuid_or_id (Union[str, int]): device uuid (str) or id (int).
+            uuid_or_id (Optional[Union[str, int]]): device uuid (str) or id (int) or None for sdk running on device.
             force (Optional[bool]): If force is True, the update lock will be overridden.
 
         Examples:
             >>> balena.models.device.reboot('8f66ec7')
         """
 
-        def __reboot():
-            data = {}
-            if force:
-                data = {"force": bool(force)}
+        path = "/v1/reboot"
+        force_body = {}
+        if force:
+            force_body = {"force": bool(force)}
 
+        if self.__should_run_on_device(uuid_or_id):
+            self.__supervisor_request("POST", path, force_body)
+            return
+
+        if uuid_or_id is None:
+            raise exceptions.LocalSupervisorNotFound()
+
+        def __reboot():
             device_id = uuid_or_id if is_id(uuid_or_id) else self.get(uuid_or_id, {"$select": "id"})["id"]
 
             return request(
                 method="POST",
-                path="/supervisor/v1/reboot",
+                path=f"/supervisor{path}",
                 settings=self.__settings,
-                body={"deviceId": device_id, "data": data},
+                body={"deviceId": device_id, "data": force_body},
             )
 
         with_supervisor_locked_error(__reboot)
 
-    def shutdown(self, uuid_or_id: Union[str, int], force: bool = False) -> None:
+    def shutdown(self, uuid_or_id: Optional[Union[str, int]] = None, force: bool = False) -> None:
         """
         Shutdown the device.
 
         Args:
-            uuid_or_id (Union[str, int]): device uuid (str) or id (int).
+            uuid_or_id (Optional[Union[str, int]]): device uuid (str) or id (int) or None for sdk running on device.
             force (Optional[bool]): If force is True, the update lock will be overridden.
 
         Examples:
             >>> balena.models.device.shutdown('8f66ec7')
         """
 
-        def __shutdown():
-            data = {}
-            if force:
-                data = {"force": bool(force)}
+        path = "/v1/shutdown"
+        force_reboot = {}
+        if force:
+            force_reboot = {"force": bool(force)}
```

Comment:
Since `force` is quite a critical flag, I'd consider hard-checking the correct value type in order to absolutely minimize accidental triggers, e.g.
```python
if not isinstance(force, bool):
    raise ValueError(f"The `force` flag must be of type bool, got {type(force)} instead.")
```

WDYT?

Reply:
great suggstion :+1: